gossip/group.go

```diff
@@ -202,13 +202,12 @@ func (g *group) addInfo(i *info) (contentsChanged bool, err error) {
 	// value of prospective info is lower, take the minimum of the two
 	// Hops values.
 	if existingInfo, ok := g.Infos[i.Key]; ok {
-		if existingInfo.Timestamp < i.Timestamp ||
-			(existingInfo.Timestamp == i.Timestamp && existingInfo.Hops > i.Hops) {
-			g.removeInternal(existingInfo)
-		} else {
-			err = util.Errorf("current group info %+v newer than proposed info %+v", existingInfo, i)
+		if i.Timestamp < existingInfo.Timestamp ||
+			(i.Timestamp == existingInfo.Timestamp && i.Hops >= existingInfo.Hops) {
```

Comment:
`i.Timestamp == existingInfo.Timestamp` and `i.Hops == existingInfo.Hops` should trigger an update? If so, add a comment to explain why.